compiler/src/dotty/tools/dotc/transform/init/Effects.scala

```diff
@@ -12,8 +12,8 @@ import core.Contexts._
 import Potentials._
 
 object Effects {
-  type Effects = Set[Effect]
-  val empty: Effects = Set.empty
+  type Effects = Vector[Effect]
```

Comment:
Is the change from Set to Vector related to checking `TypeDef`s, or is it an orthogonal improvement?

Reply:
It's orthogonal. Set is bad for concatenation, and we can afford to have duplicate entries in summaries.